frb_codegen/src/ir/ty_primitive_list.rs

```diff
@@ -2,21 +2,41 @@ use crate::ir::*;
 use convert_case::{Case, Casing};
 
 #[derive(Debug, Clone)]
-pub struct IrTypePrimitiveList {
-    pub primitive: IrTypePrimitive,
+pub enum IrTypePrimitiveList {
+    PrimitiveList {
+        primitive: IrTypePrimitive,
+    },
+    PrimitiveArray {
+        primitive: IrTypePrimitive,
+        len: usize,
+    },
+}
+
+impl IrTypePrimitiveList {
+    pub fn primitive(&self) -> &IrTypePrimitive {
+        match self {
+            IrTypePrimitiveList::PrimitiveList { primitive } => primitive,
+            IrTypePrimitiveList::PrimitiveArray { primitive, .. } => primitive,
+        }
+    }
 }
 
 impl IrTypeTrait for IrTypePrimitiveList {
     fn visit_children_types<F: FnMut(&IrType) -> bool>(&self, f: &mut F, _ir_file: &IrFile) {
-        f(&IrType::Primitive(self.primitive.clone()));
+        f(&IrType::Primitive(self.primitive().clone()));
     }
 
     fn safe_ident(&self) -> String {
-        self.dart_api_type().to_case(Case::Snake)
+        if let Self::PrimitiveArray { .. } = self {
+            self.dart_api_type().replace("List", "Array")
```

Comment:
nit: Well, maybe we can make it less hacky here... This code depends on the current fact that dart_api_type has a "List" substring in it, but that looks a bit fragile